apps/site/src/components/pages/wordpress/email-submitted-context.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+
+export type EmailSubmittedContextValue = {
+  submittedEmail: string | null;
+  setSubmittedEmail: (email: string) => void;
+};
+
+export const EmailSubmittedContext =
+  createContext<EmailSubmittedContextValue | null>(null);
+
+export const EmailSubmittedProvider = ({ children }: PropsWithChildren) => {
+  const [submittedEmail, setSubmittedEmail] = useState<string | null>(null);
+
+  const value = useMemo(
+    () => ({ submittedEmail, setSubmittedEmail }),
+    [submittedEmail, setSubmittedEmail],
+  );
+
+  return (
+    <EmailSubmittedContext.Provider value={value}>
+      {children}
+    </EmailSubmittedContext.Provider>
+  );
+};
+
+export const useEmailSubmitted = () => {
```

Comment:
Is this going to be also used in the voting footer? At the moment it seems unnecessary

Reply:
I don't think it will be needed for the voting footer, no. I'm using it to track in the whole page if an email has been submitted to disable all the other email inputs. Are you suggesting it could be a state variable on the page component passed to each component that has an email input?